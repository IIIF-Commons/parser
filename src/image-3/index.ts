export * from './types';
export * from './parser/parse-image-server-from-id';
export * from './parser/parse-image-service-request';
export * from './parser/parse-image-service-url';
export * from './parser/parse-region-parameter';
export * from './parser/parse-rotation-parameter';
export * from './parser/parse-size-parameter';
export * from './profiles/profiles';
export * from './profiles/combine-profiles';
export * from './profiles/level-to-profile';
export * from './profiles/is-level-0';
export * from './profiles/supports';
export * from './profiles/supports-custom-sizes';
export * from './profiles/image-service-supports-format';
export * from './profiles/image-service-supports-request';
export * from './serialize/image-service-request-to-string';
export * from './serialize/image-service-request-info';
export * from './serialize/region-parameter-to-string';
export * from './serialize/rotation-parameter-to-string';
export * from './serialize/size-parameter-to-string';
export * from './utilities/canonical-service-url';
export * from './utilities/create-image-service-request';
export * from './utilities/extract-fixed-size-scales';
export * from './utilities/fixed-sizes-from-scales';
export * from './utilities/get-id';
export * from './utilities/get-type';
export * from './utilities/is-image-service';