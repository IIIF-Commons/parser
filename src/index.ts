// Exports the latest
export * from './presentation-3';
