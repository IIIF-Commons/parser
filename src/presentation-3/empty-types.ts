import {
  AnnotationNormalized,
  AnnotationPageNormalized,
  CanvasNormalized,
  CollectionNormalized,
  ManifestNormalized,
  RangeNormalized,
  ResourceProviderNormalized,
} from '@iiif/presentation-3-normalized';
import { _ServiceNormalized } from './serialize';
import { EMPTY } from './utilities';

export const emptyAnnotation: AnnotationNormalized = {
  id: 'https://iiif-parser/annotation',
  type: 'Annotation',
  behavior: EMPTY,
  label: null,
  thumbnail: EMPTY,
  summary: null,
  requiredStatement: null,
  metadata: EMPTY,
  seeAlso: EMPTY,
  homepage: EMPTY,
  rendering: EMPTY,
  service: EMPTY,
  accessibility: EMPTY,
  audience: EMPTY,
  body: EMPTY,
  bodyValue: null,
  canonical: null,
  created: null,
  creator: EMPTY,
  generated: null,
  generator: EMPTY,
  modified: null,
  motivation: EMPTY,
  rights: null as any, // @todo bug? should not be array of strings.
  stylesheet: null,
  target: EMPTY,
  timeMode: undefined, // @todo bug? should be null.
  via: EMPTY,
  partOf: EMPTY,
};

export const emptyAnnotationPage: AnnotationPageNormalized = {
  id: 'https://iiif-parser/annotation-page',
  type: 'AnnotationPage',
  behavior: EMPTY,
  label: null,
  thumbnail: EMPTY,
  summary: null,
  requiredStatement: null,
  metadata: EMPTY,
  rights: null,
  provider: EMPTY,
  items: EMPTY,
  seeAlso: EMPTY,
  homepage: EMPTY,
  rendering: EMPTY,
  service: EMPTY,
};

export const emptyCanvas: CanvasNormalized = {
  id: 'https://iiif-parser/empty-canvas',
  type: 'Canvas',
  label: null,
  behavior: EMPTY,
  thumbnail: EMPTY,
  accompanyingCanvas: null,
  placeholderCanvas: null,
  summary: null,
  requiredStatement: null,
  metadata: EMPTY,
  rights: null,
  navDate: null,
  provider: EMPTY,
  items: EMPTY,
  annotations: EMPTY,
  seeAlso: EMPTY,
  homepage: EMPTY,
  partOf: EMPTY,
  rendering: EMPTY,
  service: EMPTY,
  duration: 0,
  height: 0,
  width: 0,
};

export const emptyCollection: CollectionNormalized = {
  id: 'https://iiif-parser/empty-collection',
  type: 'Collection',
  label: null,
  viewingDirection: 'left-to-right',
  behavior: EMPTY,
  thumbnail: EMPTY,
  accompanyingCanvas: null,
  placeholderCanvas: null,
  summary: null,
  requiredStatement: null,
  metadata: EMPTY,
  rights: null,
  navDate: null,
  provider: EMPTY,
  items: EMPTY,
  annotations: EMPTY,
  seeAlso: EMPTY,
  homepage: EMPTY,
  partOf: EMPTY,
  rendering: EMPTY,
  service: EMPTY,
  services: EMPTY,
};

export const emptyManifest: ManifestNormalized = {
  id: 'https://iiif-parser/empty-manifest',
  type: 'Manifest',
  annotations: EMPTY,
  behavior: EMPTY,
  homepage: EMPTY,
  items: EMPTY,
  label: null,
  metadata: EMPTY,
  navDate: null,
  provider: EMPTY,
  partOf: EMPTY,
  accompanyingCanvas: null,
  placeholderCanvas: null,
  rendering: EMPTY,
  requiredStatement: null,
  rights: null,
  seeAlso: EMPTY,
  service: EMPTY,
  services: EMPTY,
  start: null,
  structures: EMPTY,
  summary: null,
  thumbnail: EMPTY,
  viewingDirection: 'left-to-right',
};

export const emptyRange: RangeNormalized = {
  id: 'https://iiif-parser/empty-canvas',
  type: 'Range',
  label: null,
  behavior: EMPTY,
  thumbnail: EMPTY,
  accompanyingCanvas: null,
  placeholderCanvas: null,
  summary: null,
  requiredStatement: null,
  metadata: EMPTY,
  rights: null,
  navDate: null,
  provider: EMPTY,
  items: EMPTY,
  annotations: EMPTY,
  seeAlso: EMPTY,
  homepage: EMPTY,
  partOf: EMPTY,
  rendering: EMPTY,
  service: EMPTY,
  start: null,
  supplementary: null,
  viewingDirection: 'left-to-right',
};

export const emptyAgent: ResourceProviderNormalized = {
  id: 'https://iiif-parser/empty-agent',
  type: 'Agent',
  label: {},
  logo: EMPTY,
  seeAlso: EMPTY,
  homepage: EMPTY,
};

export const emptyService: _ServiceNormalized = {
  id: 'https://iiif-parser/empty-service',
  type: 'UnknownService',
} as any;
