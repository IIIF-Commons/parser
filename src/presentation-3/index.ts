export * from "./empty-types";
export * from "./normalize";
export * from "./traverse";
export * from "./serialize";
export * from "./serialize-presentation-2";
export * from "./serialize-presentation-3";
