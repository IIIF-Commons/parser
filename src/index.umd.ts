import * as Presentation2 from "./presentation-2";
import * as Presentation3 from "./presentation-3";

export default {
  Presentation2,
  Presentation3,
};
