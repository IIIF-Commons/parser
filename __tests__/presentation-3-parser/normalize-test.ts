import { normalize } from '../../src/presentation-3';

import manifestFixture from '../../fixtures/2-to-3-converted/manifests/iiif.io__api__presentation__2.1__example__fixtures__1__manifest.json';
import manifestWithStartFixture from '../../fixtures/presentation-3/start-canvas.json';

describe('normalize', () => {
  test('normalize simple manifest', () => {
    const db = normalize(manifestFixture);

    expect(db.mapping).toMatchInlineSnapshot(`
      {
        "http://iiif.io/api/presentation/2.1/example/fixtures/1/manifest.json": "Manifest",
        "http://iiif.io/api/presentation/2.1/example/fixtures/canvas/1/c1.json": "Canvas",
        "http://iiif.io/api/presentation/2.1/example/fixtures/collection.json": "ContentResource",
        "http://iiif.io/api/presentation/2.1/example/fixtures/resources/page1-full.png": "ContentResource",
        "https://example.org/uuid/1dad04b3-79c6-4a97-a831-634f2ee50a26": "AnnotationPage",
        "https://example.org/uuid/3644c005-bf7a-48d0-9fa9-1e1757bf8df1": "Annotation",
      }
    `);
  });

  test('normalize', () => {
    const manifest = () => ({
      '@context': ['http://www.w3.org/ns/anno.jsonld', 'http://iiif.io/api/presentation/{{ page.major }}/context.json'],
      id: 'https://example.org/iiif/book1/manifest',
      type: 'Manifest',
      label: { en: ['Image 1'] },
      homepage: [{ id: 'http://myhomepage.com' }],
      items: [
        {
          id: 'https://example.org/iiif/book1/canvas/p1',
          type: 'Canvas',
          height: 1800,
          width: 1200,
          items: [
            {
              id: 'https://example.org/iiif/book1/page/p1/1',
              type: 'AnnotationPage',
              items: [
                {
                  id: 'https://example.org/iiif/book1/annotation/p0001-image',
                  type: 'Annotation',
                  motivation: 'painting',
                  body: {
                    id: 'http://iiif.io/api/presentation/2.1/example/fixtures/resources/page1-full.png',
                    type: 'Image',
                    format: 'image/png',
                    height: 1800,
                    width: 1200,
                  },
                  target: 'https://example.org/iiif/book1/canvas/p1',
                },
              ],
            },
          ],
        },
      ],
    });

    const result = normalize(manifest());

    expect(result.entities).toMatchSnapshot();

    expect(result.mapping).toEqual({
      'http://iiif.io/api/presentation/2.1/example/fixtures/resources/page1-full.png': 'ContentResource',
      'http://myhomepage.com': 'ContentResource',
      'https://example.org/iiif/book1/annotation/p0001-image': 'Annotation',
      'https://example.org/iiif/book1/canvas/p1': 'Canvas',
      'https://example.org/iiif/book1/manifest': 'Manifest',
      'https://example.org/iiif/book1/page/p1/1': 'AnnotationPage',
    });

    expect(result.resource).toEqual({
      id: 'https://example.org/iiif/book1/manifest',
      type: 'Manifest',
    });
  });

  test('normalize full example from specification', () => {
    const result = normalize({
      // Metadata about this manifest file
      '@context': 'http://iiif.io/api/presentation/3/context.json',
      id: 'https://example.org/iiif/book1/manifest',
      type: 'Manifest',

      // Descriptive metadata about the object/work
      label: { en: ['Book 1'] },
      metadata: [
        {
          label: { en: ['Author'] },
          value: { none: ['Anne Author'] },
        },
        {
          label: { en: ['Published'] },
          value: {
            en: ['Paris, circa 1400'],
            fr: ['Paris, environ 1400'],
          },
        },
        {
          label: { en: ['Notes'] },
          value: {
            en: ['Text of note 1', 'Text of note 2'],
          },
        },
        {
          label: { en: ['Source'] },
          value: { none: ['<span>From: <a href="https://example.org/db/1.html">Some Collection</a></span>'] },
        },
      ],
      summary: { en: ['Book 1, written by Anne Author, published in Paris around 1400.'] },

      thumbnail: [
        {
          id: 'https://example.org/iiif/book1/page1/full/80,100/0/default.jpg',
          type: 'Image',
          format: 'image/jpeg',
          service: [
            {
              id: 'https://example.org/iiif/book1/page1',
              type: 'ImageService3',
              profile: 'level1',
            },
          ],
        },
      ],

      // Presentation Information
      viewingDirection: 'right-to-left',
      behavior: ['paged'],
      navDate: '1856-01-01T00:00:00Z',

      // Rights Information
      rights: 'http://creativecommons.org/licenses/by/4.0/',
      requiredStatement: {
        label: { en: ['Attribution'] },
        value: { en: ['Provided by Example Organization'] },
      },
      provider: [
        {
          id: 'https://example.org/about',
          type: 'Agent',
          label: { en: ['Example Organization'] },
          homepage: [
            {
              id: 'https://example.org/',
              type: 'Text',
              label: { en: ['Example Organization Homepage'] },
              format: 'text/html',
            },
          ],
          logo: [
            {
              id: 'https://example.org/service/inst1/full/max/0/default.png',
              type: 'Image',
              format: 'image/png',
              service: [
                {
                  id: 'https://example.org/service/inst1',
                  type: 'ImageService3',
                  profile: 'level2',
                },
              ],
            },
          ],
          seeAlso: [
            {
              id: 'https://data.example.org/about/us.jsonld',
              type: 'Dataset',
              format: 'application/ld+json',
              profile: 'https://schema.org/',
            },
          ],
        },
      ],

      // Links
      homepage: [
        {
          id: 'https://example.org/info/book1/',
          type: 'Text',
          label: { en: ['Home page for Book 1'] },
          format: 'text/html',
        },
      ],
      service: [
        {
          id: 'https://example.org/service/example',
          type: 'ExampleExtensionService',
          profile: 'https://example.org/docs/example-service.html',
        },
      ],
      seeAlso: [
        {
          id: 'https://example.org/library/catalog/book1.xml',
          type: 'Dataset',
          format: 'text/xml',
          profile: 'https://example.org/profiles/bibliographic',
        },
      ],
      rendering: [
        {
          id: 'https://example.org/iiif/book1.pdf',
          type: 'Text',
          label: { en: ['Download as PDF'] },
          format: 'application/pdf',
        },
      ],
      partOf: [
        {
          id: 'https://example.org/collections/books/',
          type: 'Collection',
        },
      ],
      start: {
        id: 'https://example.org/iiif/book1/canvas/p2',
        type: 'Canvas',
      },

      // List of Services, referenced from within items, structures or annotations
      services: [
        {
          '@id': 'https://example.org/iiif/auth/login',
          '@type': 'AuthCookieService1',
          profile: 'http://iiif.io/api/auth/1/login',
          label: 'Login to Example Institution',
          service: [
            {
              '@id': 'https://example.org/iiif/auth/token',
              '@type': 'AuthTokenService1',
              profile: 'http://iiif.io/api/auth/1/token',
            },
          ],
        },
      ],

      // List of Canvases
      items: [
        {
          id: 'https://example.org/iiif/book1/canvas/p1',
          type: 'Canvas',
          label: { none: ['p. 1'] },
          // ...
        },
      ],

      // structure of the resource, described with Ranges
      structures: [
        {
          id: 'https://example.org/iiif/book1/range/top',
          type: 'Range',
          // Ranges members should be included here
        },
        // Any additional top level Ranges can be included here
      ],

      // Commentary Annotations on the Manifest
      annotations: [
        {
          id: 'https://example.org/iiif/book1/annotations/p1',
          type: 'AnnotationPage',
          items: [
            // Annotations about the Manifest are included here
          ],
        },
      ],
    });

    expect(result.entities).toMatchSnapshot();

    expect(result.mapping).toEqual({
      'https://data.example.org/about/us.jsonld': 'ContentResource',
      'https://example.org/': 'ContentResource',
      'https://example.org/collections/books/': 'ContentResource',
      'https://example.org/iiif/book1.pdf': 'ContentResource',
      'https://example.org/iiif/book1/annotations/p1': 'AnnotationPage',
      'https://example.org/iiif/book1/canvas/p1': 'Canvas',
      'https://example.org/iiif/book1/canvas/p2': 'Canvas',
      'https://example.org/iiif/book1/manifest': 'Manifest',
      'https://example.org/iiif/book1/page1/full/80,100/0/default.jpg': 'ContentResource',
      'https://example.org/iiif/book1/range/top': 'Range',
      'https://example.org/info/book1/': 'ContentResource',
      'https://example.org/library/catalog/book1.xml': 'ContentResource',
      'https://example.org/service/inst1/full/max/0/default.png': 'ContentResource',
      'https://example.org/about': 'Agent',
    });

    expect(result.resource).toEqual({
      id: 'https://example.org/iiif/book1/manifest',
      type: 'Manifest',
    });
  });

  test('normalize manifest with start property', () => {
    const db = normalize(manifestWithStartFixture);
    expect(
      (db.entities.Canvas as any)['https://iiif.io/api/cookbook/recipe/0202-start-canvas/canvas/p2']
    ).toMatchObject({
      id: 'https://iiif.io/api/cookbook/recipe/0202-start-canvas/canvas/p2',
      type: 'Canvas',
      label: {
        en: ['Frontispiece'],
      },
      width: 3186,
      height: 4612,
    });
  });
});
