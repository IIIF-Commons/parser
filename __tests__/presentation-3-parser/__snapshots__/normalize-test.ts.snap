// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`normalize normalize 1`] = `
Object {
  "Agent": Object {},
  "Annotation": Object {
    "https://example.org/iiif/book1/annotation/p0001-image": Object {
      "body": Array [
        Object {
          "id": "http://iiif.io/api/presentation/2.1/example/fixtures/resources/page1-full.png",
          "type": "ContentResource",
        },
      ],
      "id": "https://example.org/iiif/book1/annotation/p0001-image",
      "motivation": Array [
        "painting",
      ],
      "target": "https://example.org/iiif/book1/canvas/p1",
      "type": "Annotation",
    },
  },
  "AnnotationCollection": Object {},
  "AnnotationPage": Object {
    "https://example.org/iiif/book1/page/p1/1": Object {
      "behavior": Array [],
      "homepage": Array [],
      "id": "https://example.org/iiif/book1/page/p1/1",
      "items": Array [
        Object {
          "id": "https://example.org/iiif/book1/annotation/p0001-image",
          "type": "Annotation",
        },
      ],
      "label": null,
      "logo": Array [],
      "metadata": Array [],
      "motivation": null,
      "provider": Array [],
      "rendering": Array [],
      "requiredStatement": null,
      "rights": null,
      "seeAlso": Array [],
      "service": Array [],
      "summary": null,
      "thumbnail": Array [],
      "type": "AnnotationPage",
    },
  },
  "Canvas": Object {
    "https://example.org/iiif/book1/canvas/p1": Object {
      "accompanyingCanvas": null,
      "annotations": Array [],
      "behavior": Array [],
      "duration": 0,
      "height": 1800,
      "homepage": Array [],
      "id": "https://example.org/iiif/book1/canvas/p1",
      "items": Array [
        Object {
          "id": "https://example.org/iiif/book1/page/p1/1",
          "type": "AnnotationPage",
        },
      ],
      "label": null,
      "logo": Array [],
      "metadata": Array [],
      "motivation": null,
      "navDate": null,
      "partOf": Array [],
      "placeholderCanvas": null,
      "posterCanvas": null,
      "provider": Array [],
      "rendering": Array [],
      "requiredStatement": null,
      "rights": null,
      "seeAlso": Array [],
      "service": Array [],
      "summary": null,
      "thumbnail": Array [],
      "type": "Canvas",
      "width": 1200,
    },
  },
  "Collection": Object {},
  "ContentResource": Object {
    "http://iiif.io/api/presentation/2.1/example/fixtures/resources/page1-full.png": Object {
      "format": "image/png",
      "height": 1800,
      "id": "http://iiif.io/api/presentation/2.1/example/fixtures/resources/page1-full.png",
      "type": "Image",
      "width": 1200,
    },
    "http://myhomepage.com": Object {
      "id": "http://myhomepage.com",
      "type": "ContentResource",
    },
  },
  "Manifest": Object {
    "https://example.org/iiif/book1/manifest": Object {
      "@context": Array [
        "http://www.w3.org/ns/anno.jsonld",
        "http://iiif.io/api/presentation/{{ page.major }}/context.json",
      ],
      "accompanyingCanvas": null,
      "annotations": Array [],
      "behavior": Array [],
      "homepage": Array [
        Object {
          "id": "http://myhomepage.com",
          "type": "ContentResource",
        },
      ],
      "id": "https://example.org/iiif/book1/manifest",
      "items": Array [
        Object {
          "id": "https://example.org/iiif/book1/canvas/p1",
          "type": "Canvas",
        },
      ],
      "label": Object {
        "en": Array [
          "Image 1",
        ],
      },
      "logo": Array [],
      "metadata": Array [],
      "motivation": null,
      "navDate": null,
      "partOf": Array [],
      "placeholderCanvas": null,
      "posterCanvas": null,
      "provider": Array [],
      "rendering": Array [],
      "requiredStatement": null,
      "rights": null,
      "seeAlso": Array [],
      "service": Array [],
      "services": Array [],
      "start": null,
      "structures": Array [],
      "summary": null,
      "thumbnail": Array [],
      "type": "Manifest",
      "viewingDirection": "left-to-right",
    },
  },
  "Range": Object {},
  "Selector": Object {},
  "Service": Object {},
}
`;

exports[`normalize normalize full example from specification 1`] = `
Object {
  "Agent": Object {
    "https://example.org/about": Object {
      "homepage": Array [
        Object {
          "id": "https://example.org/",
          "type": "ContentResource",
        },
      ],
      "id": "https://example.org/about",
      "label": Object {
        "en": Array [
          "Example Organization",
        ],
      },
      "logo": Array [
        Object {
          "id": "https://example.org/service/inst1/full/max/0/default.png",
          "type": "ContentResource",
        },
      ],
      "seeAlso": Array [
        Object {
          "id": "https://data.example.org/about/us.jsonld",
          "type": "ContentResource",
        },
      ],
      "type": "Agent",
    },
  },
  "Annotation": Object {},
  "AnnotationCollection": Object {},
  "AnnotationPage": Object {
    "https://example.org/iiif/book1/annotations/p1": Object {
      "behavior": Array [],
      "homepage": Array [],
      "id": "https://example.org/iiif/book1/annotations/p1",
      "items": Array [],
      "label": null,
      "logo": Array [],
      "metadata": Array [],
      "motivation": null,
      "provider": Array [],
      "rendering": Array [],
      "requiredStatement": null,
      "rights": null,
      "seeAlso": Array [],
      "service": Array [],
      "summary": null,
      "thumbnail": Array [],
      "type": "AnnotationPage",
    },
  },
  "Canvas": Object {
    "https://example.org/iiif/book1/canvas/p1": Object {
      "accompanyingCanvas": null,
      "annotations": Array [],
      "behavior": Array [],
      "duration": 0,
      "height": 0,
      "homepage": Array [],
      "id": "https://example.org/iiif/book1/canvas/p1",
      "items": Array [],
      "label": Object {
        "none": Array [
          "p. 1",
        ],
      },
      "logo": Array [],
      "metadata": Array [],
      "motivation": null,
      "navDate": null,
      "partOf": Array [],
      "placeholderCanvas": null,
      "posterCanvas": null,
      "provider": Array [],
      "rendering": Array [],
      "requiredStatement": null,
      "rights": null,
      "seeAlso": Array [],
      "service": Array [],
      "summary": null,
      "thumbnail": Array [],
      "type": "Canvas",
      "width": 0,
    },
    "https://example.org/iiif/book1/canvas/p2": Object {
      "accompanyingCanvas": null,
      "annotations": Array [],
      "behavior": Array [],
      "duration": 0,
      "height": 0,
      "homepage": Array [],
      "id": "https://example.org/iiif/book1/canvas/p2",
      "items": Array [],
      "label": null,
      "logo": Array [],
      "metadata": Array [],
      "motivation": null,
      "navDate": null,
      "partOf": Array [],
      "placeholderCanvas": null,
      "posterCanvas": null,
      "provider": Array [],
      "rendering": Array [],
      "requiredStatement": null,
      "rights": null,
      "seeAlso": Array [],
      "service": Array [],
      "summary": null,
      "thumbnail": Array [],
      "type": "Canvas",
      "width": 0,
    },
  },
  "Collection": Object {},
  "ContentResource": Object {
    "https://data.example.org/about/us.jsonld": Object {
      "format": "application/ld+json",
      "id": "https://data.example.org/about/us.jsonld",
      "profile": "https://schema.org/",
      "type": "Dataset",
    },
    "https://example.org/": Object {
      "format": "text/html",
      "id": "https://example.org/",
      "label": Object {
        "en": Array [
          "Example Organization Homepage",
        ],
      },
      "type": "Text",
    },
    "https://example.org/collections/books/": Object {
      "id": "https://example.org/collections/books/",
      "type": "Collection",
    },
    "https://example.org/iiif/book1.pdf": Object {
      "format": "application/pdf",
      "id": "https://example.org/iiif/book1.pdf",
      "label": Object {
        "en": Array [
          "Download as PDF",
        ],
      },
      "type": "Text",
    },
    "https://example.org/iiif/book1/page1/full/80,100/0/default.jpg": Object {
      "format": "image/jpeg",
      "id": "https://example.org/iiif/book1/page1/full/80,100/0/default.jpg",
      "service": Array [
        Object {
          "id": "https://example.org/iiif/book1/page1",
          "profile": "level1",
          "type": "ImageService3",
        },
      ],
      "type": "Image",
    },
    "https://example.org/info/book1/": Object {
      "format": "text/html",
      "id": "https://example.org/info/book1/",
      "label": Object {
        "en": Array [
          "Home page for Book 1",
        ],
      },
      "type": "Text",
    },
    "https://example.org/library/catalog/book1.xml": Object {
      "format": "text/xml",
      "id": "https://example.org/library/catalog/book1.xml",
      "profile": "https://example.org/profiles/bibliographic",
      "type": "Dataset",
    },
    "https://example.org/service/inst1/full/max/0/default.png": Object {
      "format": "image/png",
      "id": "https://example.org/service/inst1/full/max/0/default.png",
      "service": Array [
        Object {
          "id": "https://example.org/service/inst1",
          "profile": "level2",
          "type": "ImageService3",
        },
      ],
      "type": "Image",
    },
  },
  "Manifest": Object {
    "https://example.org/iiif/book1/manifest": Object {
      "@context": "http://iiif.io/api/presentation/3/context.json",
      "accompanyingCanvas": null,
      "annotations": Array [
        Object {
          "id": "https://example.org/iiif/book1/annotations/p1",
          "type": "AnnotationPage",
        },
      ],
      "behavior": Array [
        "paged",
      ],
      "homepage": Array [
        Object {
          "id": "https://example.org/info/book1/",
          "type": "ContentResource",
        },
      ],
      "id": "https://example.org/iiif/book1/manifest",
      "items": Array [
        Object {
          "id": "https://example.org/iiif/book1/canvas/p1",
          "type": "Canvas",
        },
      ],
      "label": Object {
        "en": Array [
          "Book 1",
        ],
      },
      "logo": Array [],
      "metadata": Array [
        Object {
          "label": Object {
            "en": Array [
              "Author",
            ],
          },
          "value": Object {
            "none": Array [
              "Anne Author",
            ],
          },
        },
        Object {
          "label": Object {
            "en": Array [
              "Published",
            ],
          },
          "value": Object {
            "en": Array [
              "Paris, circa 1400",
            ],
            "fr": Array [
              "Paris, environ 1400",
            ],
          },
        },
        Object {
          "label": Object {
            "en": Array [
              "Notes",
            ],
          },
          "value": Object {
            "en": Array [
              "Text of note 1",
              "Text of note 2",
            ],
          },
        },
        Object {
          "label": Object {
            "en": Array [
              "Source",
            ],
          },
          "value": Object {
            "none": Array [
              "<span>From: <a href=\\"https://example.org/db/1.html\\">Some Collection</a></span>",
            ],
          },
        },
      ],
      "motivation": null,
      "navDate": "1856-01-01T00:00:00Z",
      "partOf": Array [
        Object {
          "id": "https://example.org/collections/books/",
          "type": "ContentResource",
        },
      ],
      "placeholderCanvas": null,
      "posterCanvas": null,
      "provider": Array [
        Object {
          "id": "https://example.org/about",
          "type": "Agent",
        },
      ],
      "rendering": Array [
        Object {
          "id": "https://example.org/iiif/book1.pdf",
          "type": "ContentResource",
        },
      ],
      "requiredStatement": Object {
        "label": Object {
          "en": Array [
            "Attribution",
          ],
        },
        "value": Object {
          "en": Array [
            "Provided by Example Organization",
          ],
        },
      },
      "rights": "http://creativecommons.org/licenses/by/4.0/",
      "seeAlso": Array [
        Object {
          "id": "https://example.org/library/catalog/book1.xml",
          "type": "ContentResource",
        },
      ],
      "service": Array [
        Object {
          "id": "https://example.org/service/example",
          "profile": "https://example.org/docs/example-service.html",
          "type": "ExampleExtensionService",
        },
      ],
      "services": Array [
        Object {
          "@id": "https://example.org/iiif/auth/login",
          "@type": "AuthCookieService1",
          "label": "Login to Example Institution",
          "profile": "http://iiif.io/api/auth/1/login",
          "service": Array [
            Object {
              "@id": "https://example.org/iiif/auth/token",
              "@type": "AuthTokenService1",
              "profile": "http://iiif.io/api/auth/1/token",
            },
          ],
        },
      ],
      "start": Object {
        "selector": undefined,
        "source": Object {
          "id": "https://example.org/iiif/book1/canvas/p2",
          "type": "Canvas",
        },
        "type": "SpecificResource",
      },
      "structures": Array [
        Object {
          "id": "https://example.org/iiif/book1/range/top",
          "type": "Range",
        },
      ],
      "summary": Object {
        "en": Array [
          "Book 1, written by Anne Author, published in Paris around 1400.",
        ],
      },
      "thumbnail": Array [
        Object {
          "id": "https://example.org/iiif/book1/page1/full/80,100/0/default.jpg",
          "type": "ContentResource",
        },
      ],
      "type": "Manifest",
      "viewingDirection": "right-to-left",
    },
  },
  "Range": Object {
    "https://example.org/iiif/book1/range/top": Object {
      "accompanyingCanvas": null,
      "annotations": Array [],
      "behavior": Array [],
      "homepage": Array [],
      "id": "https://example.org/iiif/book1/range/top",
      "items": Array [],
      "label": null,
      "logo": Array [],
      "metadata": Array [],
      "motivation": null,
      "navDate": null,
      "partOf": Array [],
      "placeholderCanvas": null,
      "posterCanvas": null,
      "provider": Array [],
      "rendering": Array [],
      "requiredStatement": null,
      "rights": null,
      "seeAlso": Array [],
      "service": Array [],
      "start": null,
      "summary": null,
      "supplementary": null,
      "thumbnail": Array [],
      "type": "Range",
      "viewingDirection": "left-to-right",
    },
  },
  "Selector": Object {},
  "Service": Object {},
}
`;
